### PR TITLE
remove checking for toc items

### PR DESCRIPTION
**Pull request by OsmondJiang**
there is no reason to check the length of toc items need to be bigger than 0

fix: https://ceapex.visualstudio.com/Engineering/_workitems/edit/88155

### CODE REVIEW

**Comment by yufeih** [comment]
@live1206 , where is the original TOC validation spec and does it mentioned anything about this validation?

**Comment by live1206** [comment]
> @live1206 , where is the original TOC validation spec and does it mentioned anything about this validation?

https://review.docs.microsoft.com/en-us/new-hope/specs/ops/toc_schema_validation?branch=master
Seems like there is no requirement of minimum length 1, we can remove it

File: docs/specs/toc.yml [hunk 1]
```diff
@@ -986,7 +986,6 @@ inputs:
       tocHref: 
       href: f1/b.md
       additionalProperty: a
-      items: []
```
Comment (live1206) [inline docs/specs/toc.yml]: keep this line to verify no error log

File: src/docfx/build/toc/TableOfContentsItem.cs [hunk 2]
```diff
@@ -31,7 +30,6 @@ public class TableOfContentsItem
 
         public List<string> Monikers { get; set; }
 
-        [MinLength(1)]
         public List<TableOfContentsItem> Items { get; set; } = new List<TableOfContentsItem>();
```
Comment (yufeih) [inline src/docfx/build/toc/TableOfContentsItem.cs]: also remove from `TableOfContentsModel`

File: docs/specs/toc.yml (hunk 1 above)
Comment (OsmondJiang) [reply docs/specs/toc.yml]: good idea

File: src/docfx/build/toc/TableOfContentsItem.cs (hunk 2 above)
Comment (OsmondJiang) [reply src/docfx/build/toc/TableOfContentsItem.cs]: nice catch!

**Pull request merged**